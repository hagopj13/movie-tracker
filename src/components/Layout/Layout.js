// @flow
import React, { useState } from 'react';
import type { Node } from 'react';

import Header from 'components/Header/Header';
import Drawer from 'components/Drawer/Drawer';
import DialogRoot from 'components/Dialog/Root/DialogRoot';

type Props = {
  children: Node,
};

const Layout = (props: Props) => {
  const { children } = props;

  const [isDrawerOpen, setIsDrawerOpen] = useState(false);

  const handleOpenDrawer = () => {
    setIsDrawerOpen(true);
  };

  const handleCloseDrawer = () => {
    setIsDrawerOpen(false);
  };

  return (
    <>
      <Header onOpenDrawer={handleOpenDrawer} />
      <Drawer isOpen={isDrawerOpen} onClose={handleCloseDrawer} />
      <DialogRoot />
      {children}
    </>
  );
};

export default Layout;
