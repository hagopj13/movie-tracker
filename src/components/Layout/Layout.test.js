import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import Layout from './Layout';

describe('Layout component', () => {
  let shallow;
  const ChildComponent = () => <div>Child component</div>;
  let wrapper;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    wrapper = shallow(
      <Layout>
        <ChildComponent />
      </Layout>,
    );
  });

  it('should render the Layout component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the child component', () => {
    expect(wrapper.exists(ChildComponent)).toBe(true);
  });
});
