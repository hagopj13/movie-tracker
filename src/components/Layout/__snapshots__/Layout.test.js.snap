// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should render the Layout component correctly 1`] = `
<Fragment>
  <Header
    onOpenDrawer={[Function]}
  />
  <Drawer
    isOpen={false}
    onClose={[Function]}
  />
  <Connect(DialogRoot) />
  <ChildComponent />
</Fragment>
`;
